### PR TITLE
Http client support

### PR DESCRIPTION
**Pull request by reavessm**
Support for testing rest endpoint via sniprun

Following [http file syntax](https://www.jetbrains.com/help/idea/exploring-http-syntax.html)

### CODE REVIEW

**Comment by reavessm** [comment]
At least the basics works:

image

**Comment by michaelb** [comment]
Nice idea! Thank for contributing!

I'll do a more in-depth review asap

**Comment by reavessm** [comment]
Thanks!  I'll keep this in draft status until I get some of the other http verbs implemented but I'd love feedback, especially on other trait functions and dependency/feature related stuff

**Comment by reavessm** [comment]
The http parser doesn't curently implement variable substitution, so we will need to upgrade the dependency version once https://github.com/Laeri/http-rest-file/pull/1 gets merged if we want that feature

File: Cargo.toml [hunk 1]
```diff
@@ -12,20 +12,24 @@ ignore_in_ci = []
...
 
 # jupyter-client = { git = "https://gitlab.com/srwalker101/rust-jupyter-client.git", branch = "dev" }
 
 
 # Python3 interpreter
-unindent="0.2"
+unindent = "0.2"
+
+# Http interpreter
+ureq = { version = "2", features = ["json"] }
```
[the remark this answers is not in the document]
Comment (reavessm) [reply Cargo.toml:34]: It's supposed to be quicker to build (less dependencies) and have a simpler api.  If anything it has less features but runs leaner.

File: src/interpreters/Http_original.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            let url = match line.target {
+                RequestTarget::Absolute { uri } => uri,
+                RequestTarget::RelativeOrigin { uri } => uri,
+                _ => return Err(SniprunError::CustomError(format!("Invalid url"))),
+            };
+
+            let resp = match line.method {
+                WithDefault::Some(HttpMethod::GET) => ureq::get(&url).call(),
+                WithDefault::Some(HttpMethod::POST) => ureq::post(&url).send_string(&req.body.to_string()),
+                _ => return Err(SniprunError::CustomError(format!("Invalid method"))),
```
[the remark this answers is not in the document]
Comment (reavessm) [reply src/interpreters/Http_original.rs]: You technically can add custom methods in the http file spec so I'll make it more clear that we're just not supporting those.

File: src/interpreters/Http_original.rs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                 _ => return Err(SniprunError::CustomError(format!("Invalid method"))),
+            };
+
+            for header in req.headers.into_iter() {
+                r = r.set(&header.key, &header.value);
+            }
+
+            match r.send(Cursor::new(req.body.to_string())) {
+                Ok(resp) => match resp.into_string() {
+                    Ok(text) => {
```
[the remark this answers is not in the document]
Comment (michaelb) [reply src/interpreters/Http_original.rs]: Actually nevermind, the 2nd part of my comment was incorrect
You don't need  error_truncate stuff

**Comment by michaelb** [comment]
Hi @reavessm , I 'fixed' a modification you made due to an incorrect comment of mine (error_truncate stuff, sorry). Do you agree with the latest commit on the 'http' branch I just pushed ?

**Comment by reavessm** [comment]
Oh, I like that!  Thank  you!  That really cleans it up.  I'll add that to this PR.  I'm going to reopen this for review.  

I also think we shouldn't rerun anything in case somethings fails.  I can see a situation where you have three REST calls that you want to execute sequentially and if the second one fails, you don't want the third to run.  Especially considering this isn't naming any of the requests, I think sequential order is important

**Comment by michaelb** [comment]
> I also think we shouldn't rerun anything in case somethings fails.

Fair enough

**Comment by michaelb** [comment]
@reavessm Are you satisfied with the state of your contribution ? 
As for me, this is good :-)

However, I'd like to do my own dev merge and add some cosmetics (add to readme and doc) about HTTP support, so I can't give you permission to merge into master anytime you want via the review system. (Because of how most vim plugin manager works, the master branch has to be carefully synchronized with sniprun's binary releases)

**Comment by reavessm** [comment]
@michaelb Yep!  I can change this to be based on the dev branch.  And if you need any help with docs I don't mind helping

**Pull request merged**